dialog/file.go

```diff
@@ -365,6 +382,30 @@ func (f *fileDialog) setSelected(file *fileDialogItem) {
 	}
 }
 
+func (f *fileDialog) setView(view viewLayout) {
+	f.view = view
+	if f.view == gridView {
+		f.files = fyne.NewContainerWithLayout(layout.NewGridWrapLayout(fyne.NewSize(fileIconCellWidth,
+			fileIconSize+theme.Padding()+fileTextSize)),
+		)
+	} else {
+		f.files = fyne.NewContainerWithLayout(layout.NewVBoxLayout())
+	}
+	if f.dir != nil {
+		f.refreshDir(f.dir)
+	}
+	f.filesWrapper.Objects = []fyne.CanvasObject{f.files}
+	f.filesWrapper.Refresh()
+}
+
+func (f *fileDialog) toggleView() {
+	if f.view == gridView {
+		f.setView(listView)
+	} else {
+		f.setView(gridView)
+	}
```

Comment:
Isn't this logic reversed?

Reply:
No, it's meant to toggle (switch) 
If we have A then B, if we have B then A.
If it's meant to be the other way around we won't need the if statement in the first place

Reply:
That seems a little bit contie